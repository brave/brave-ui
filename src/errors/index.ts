export { default as NullThemeProviderError } from './null-theme-provider-error'
export { default as UnknownPropValueError } from './unknown-prop-value-error'
