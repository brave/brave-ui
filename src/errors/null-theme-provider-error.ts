export default class NullThemeProviderError extends Error {
  name = 'NullThemeProviderError'

  constructor () {
    super('Theme is not defined, add <ThemeProvider /> to component tree')
  }
}
