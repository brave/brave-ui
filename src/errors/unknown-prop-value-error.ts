export default class UnknownPropValueError extends Error {
  name = 'UnknownPropValueError'

  constructor (name: string, value?: any) {
    super(`Unknown component ${name}: ${value}`)
  }
}
