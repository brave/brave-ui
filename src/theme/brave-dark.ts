import ITheme from './theme-interface'
import colors from './colors'
import defaultTheme from './brave-default'

const darkTheme: ITheme = {
  ...defaultTheme,
  name: 'Brave Dark',
  color: {
    ...defaultTheme.color,
    contextMenuBackground: colors.black,
    contextMenuForeground: colors.white,
    defaultControl: colors.white,
    defaultButtonBorder: colors.grey400
  }
}

export default darkTheme
