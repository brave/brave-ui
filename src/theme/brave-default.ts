import colors from './palette'
import ITheme from './theme-interface'

// Define colors, fonts, and sizes by purpose.
// Keys should not describe the value, but what they are to be used for.

const theme: ITheme = {
  color: {
    brandBrave: colors.orange400,
    brandBat: colors.blurple400,
    // brave controls
    brandBraveInteracting: colors.orange500,
    brandBraveActive: colors.orange200,
    brandBraveLight: colors.orange000,
    // bat controls
    brandBatInteracting: colors.blurple500,
    brandBatActive: colors.blurple200,
    // regular controls
    defaultControl: colors.grey800,
    defaultControlInteracting: colors.black,
    defaultControlActive: colors.grey700,
    // warning controls
    warn: colors.red500,
    warnInteracting: colors.red600,
    warnActive: colors.red700,
    // subtle controls
    subtle: colors.grey400,
    subtleInteracting: colors.grey500,
    subtleActive: colors.grey600,
    // disabled controls
    disabled: colors.grey300,
    // backgrounds
    primaryBackground: colors.white,
    secondaryBackground: colors.grey400
  },
  fontFamily: {
    heading: 'Poppins, sans-serif',
    body: 'Muli, sans-serif',
    system: '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif'
  }
}

export default theme
