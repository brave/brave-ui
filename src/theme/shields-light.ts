import ITheme from './theme-interface'
import defaultTheme from './brave-default'
import colors from './colors'

const shieldsLightTheme: ITheme = {
  ...defaultTheme,
  name: 'Shields Light',
  color: {
    ...defaultTheme.color,
    panelBackgroundSecondary: colors.neutral000
  }
}

export default shieldsLightTheme
