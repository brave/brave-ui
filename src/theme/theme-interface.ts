import { ThemedStyledProps } from 'styled-components'

export default interface IThemeProps {
  name: string,
  palette: { [key: string]: string }
  color: {
    contextMenuBackground: string
    contextMenuForeground: string
    brandBraveSubtleBackground: string
    brandBraveHover: string
    brandBraveMain: string
    brandBravePressed: string
    brandBraveOutline: string
    brandBrave: string
    brandBraveActive: string
    brandBatSubtleBackground: string
    brandBatHover: string
    brandBatMain: string
    brandBatPressed: string
    brandBatOutline: string
    brandBat: string
    brandBatInteracting: string
    brandBatActive: string
    detailDescription: string
    disabled: string
    disabledResourceBlocked: string
    primaryBackground: string
    secondaryBackground: string
    modalOverlayBackground: string
    defaultButtonText: string
    defaultButtonBorder: string
    defaultControl: string
    defaultControlInteracting: string
    defaultControlActive: string
    warn: string
    warnInteracting: string
    warnActive: string
    subtle: string
    subtleBackground: string
    subtleExclude: string
    subtleInteracting: string
    subtleActive: string
    text: string
    panelBackground: string
    panelBackgroundSecondary: string
    lionLogo: string
    inputBorder: string
    separatorLine: string
  }
  fontFamily: {
    heading: string
    body: string
    system: string
  }
}

export type BraveThemedStyledProps<T> = ThemedStyledProps<T, IThemeProps>
