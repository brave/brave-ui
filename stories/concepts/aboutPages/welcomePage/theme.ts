/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  welcomePage: {
    fontFamily: '"Poppins", sans-serif',
    alignItems: 'center',
    justifyContent: 'center'
  },
  waveBackground: {
    height: '-webkit-fill-available',
    width: '-webkit-fill-available'
  },
  panel: {
    backgroundColor: 'rgba(255,255,255,0.95)',
    maxWidth: '600px',
    minHeight: '580px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    padding: '50px 60px'
  },
  braveLogo: {
    width: '90px'
  },
  paymentsImage: {
    width: '230px'
  },
  importImage: {
    width: '215px'
  },
  shieldsImage: {
    width: '185px'
  },
  featuresImage: {
    width: '300px'
  },
  title: {
    fontFamily: '"Poppins", sans-serif',
    fontSize: '32px',
    color: '#212121',
    margin: '30px 0 0',
    textAlign: 'center',
    lineHeight: '44px'
  },
  text: {
    fontFamily: '"Muli", sans-serif',
    fontSize: '18px',
    color: '#76777A',
    lineHeight: '34px',
    textAlign: 'center',
    margin: '20px 0'
  },
  mainButton: {
    textTransform: 'uppercase'
  },
  sideButton: {
    textTransform: 'uppercase',
    fontWeight: '500'
  },
  arrow: {
    display: 'inline-block',
    verticalAlign: 'middle'
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: '1',
    marginBottom: '50px'
  },
  skip: {
    color: '#76777A',
    hoverColor: '#8B8A8E',
    textDecoration: 'underline',
    fontWeight: '300'
  },
  footer: {
    gridGap: '0',
    padding: '0'
  },
  footerColumnLeft: {
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  footerColumnCenter: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  footerColumnRight: {
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  bulletActive: {
    color: '#FB542B',
    hoverColor: '#FB542B',
    padding: '0 7px',
    fontSize: '40px'
  },
  bullet: {
    color: '#76777A',
    hoverColor: '#8B8A8E',
    padding: '0 7px',
    fontSize: '40px'
  }
} as any

export default theme
