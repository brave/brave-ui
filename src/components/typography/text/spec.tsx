/* This Source Code Form is subject to the terms of the Mozilla Public
* License, v. 2.0. If a copy of the MPL was not distributed with this file,
* You can obtain one at http://mozilla.org/MPL/2.0/.
*/

import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import 'jest-styled-components'

import { TestThemeProvider } from '../../../theme'
import { NullThemeProviderError, UnknownPropValueError } from '../../../errors'
import Text, { Props, Size, Type } from './index'

describe('Text tests', () => {
  const baseComponent = (props: Props = {}) => <Text {...props}>A browser with your interest at heart</Text>
  const baseComponentWithThemeProvider = (props: Props = {}) => <TestThemeProvider>
    {baseComponent(props)}
  </TestThemeProvider>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const defaultTree = create(baseComponentWithThemeProvider()).toJSON()
      const spanTagTree = create(baseComponentWithThemeProvider({ as: 'span' })).toJSON()
      const withOtherClassNameTree = create(baseComponentWithThemeProvider({ className: 'test' })).toJSON()

      expect(defaultTree).toMatchSnapshot()
      expect(spanTagTree).toMatchSnapshot()
      expect(withOtherClassNameTree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponentWithThemeProvider())
      const assertion = wrapper.find(Text.displayName)

      expect(assertion).toHaveLength(1)
    })
  })

  describe('component styling', () => {
    const sizes: [Size, number][] = [
      ['00', 10],
      ['01', 11],
      ['02', 12],
      ['03', 14],
      ['04', 16],
      ['05', 18],
      ['06', 22],
      ['07', 28],
      ['08', 38],
      ['09', 48],
      ['10', 60],
      ['11', 76]
    ]
    const types: [Type, string][] = [['body', 'Muli'], ['display', 'Poppins']]

    sizes.forEach(([size, value]) => {
      it(`should contain a rule "font-size: ${value}px" when size === "${size}"`, () => {
        const tree = create(baseComponentWithThemeProvider({ size })).toJSON()

        expect(tree).toHaveStyleRule('font-size', `${value}px`)
      })
    })

    types.forEach(([type, fontName]) => {
      it(`should contain a rule "font-family: ${fontName},sans-serif" when type === "${type}"`, () => {
        const tree = create(baseComponentWithThemeProvider({ type })).toJSON()

        expect(tree).toHaveStyleRule('font-family', `${fontName},sans-serif`)
      })
    })
  })

  describe('component error', () => {
    it('should throw an exception if passed an invalid size', () => {
      const size = '15' as Size
      const createComponent = () => create(baseComponentWithThemeProvider({ size }))

      expect(createComponent).toThrow(UnknownPropValueError)
    })

    it('should throw an exception if ThemeProvider is not define', () => {
      const createComponent = () => create(baseComponent())

      expect(createComponent).toThrow(NullThemeProviderError)
    })
  })
})
