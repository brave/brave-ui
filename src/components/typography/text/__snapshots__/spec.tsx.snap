// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text tests basic tests matches the snapshot 1`] = `
.c0 {
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Muli,sans-serif;
}

.c1 {
  font-size: 16px;
}

<div
  className="c0 c1"
  type="body"
>
  A browser with your interest at heart
</div>
`;

exports[`Text tests basic tests matches the snapshot 2`] = `
.c0 {
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Muli,sans-serif;
}

.c1 {
  font-size: 16px;
}

<span
  className="c0 c1"
  type="body"
>
  A browser with your interest at heart
</span>
`;

exports[`Text tests basic tests matches the snapshot 3`] = `
.c0 {
  line-height: 1.5;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: Muli,sans-serif;
}

.c1 {
  font-size: 16px;
}

<div
  className="c0 c1 test"
  type="body"
>
  A browser with your interest at heart
</div>
`;
